apis/network/v1beta1/openstackipset_types.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Network Type
+type Network struct {
+	// +kubebuilder:validation:Required
+	// Network Name
+	Name string `json:"name"`
+
+	// +kubebuilder:validation:Required
+	// Subnet Name
+	SubnetName string `json:"subnetName"`
+
+	// +kubebuilder:validation:Optional
+	// Fixed Ip
+	FixedIP string `json:"fixedIP,omitempty"`
+}
+
+// OpenStackIPSetSpec defines the desired state of OpenStackIPSet
+type OpenStackIPSetSpec struct {
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=false
+	// VIP flag to indicate ipset is a request for a VIP
+	VIP bool `json:"vip"`
```

Comment:
not sure if we need this for podified ctlplane. in OSPDO it was required to generate the needed env files. now I think we could just request an IP, or did I miss a use case?

Reply:
I added that to distinguish, if there is a OpenStackIPSet request for a set of vips i.e not associated with a node. If there is no requirement I can remove it later.

Reply:
sounds good